packages/mini-browser/src/browser/mini-browser-guard.ts

```diff
@@ -0,0 +1,114 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { MessageService } from '@theia/core';
+import { PreferenceService, PreferenceScope } from '@theia/core/lib/browser';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { MiniBrowserPreferences, IMiniBrowserPreferences } from './mini-browser-preferences';
+import { MiniBrowserConfiguration } from './mini-browser-configuration';
+
+/**
+ * Checks for known security issues with the mini-browser.
+ * Can be controlled through preferences.
+ */
+@injectable()
+export class MiniBrowserGuard {
+
+    @inject(MessageService)
+    protected messageService: MessageService;
+
+    @inject(PreferenceService)
+    protected preferenceService: PreferenceService;
+
+    @inject(MiniBrowserConfiguration)
+    protected miniBrowserConfiguration: MiniBrowserConfiguration;
+
+    @inject(MiniBrowserPreferences)
+    protected miniBrowserPreferences: MiniBrowserPreferences;
+
+    async onSetHostPattern(hostPattern: string): Promise<void> {
+        await this.miniBrowserPreferences.ready;
+        if (this.miniBrowserPreferences['mini-browser.warnIfUnsecure']) {
+            if (this.isHostPatternUnsecure(hostPattern)) {
+                this.messageService.warn(
+                    '`mini-browser` is currently configured to serve `file:` resources on the same origin as the application, this is known to be unsecure. ' +
```

Comment:
nitpick: "known" -> "thought"?

Reply:
I think we should add a link to the vscode webviews author's blog post, unless we have a better link. This way we provide what info we have. We can't do more than that!

Reply:
https://blog.mattbierner.com/vscode-webview-web-learnings/